pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go

```diff
@@ -255,34 +255,45 @@ func semanticIngressEquals(desired, existing *netv1.Ingress) bool {
 }
 
 func (r *RawIngressReconciler) Reconcile(isvc *v1beta1api.InferenceService) error {
-	ingress, err := createRawIngress(r.scheme, isvc, r.ingressConfig, r.client)
-	if ingress == nil {
-		return nil
+	var err error
+	isInternal := false
+	// disable ingress creation if service is labelled with cluster local or kserve domain is cluster local
+	if val, ok := isvc.Labels[constants.NetworkVisibility]; ok && val == constants.ClusterLocalVisibility {
```

Comment:
This constant `constants.NetworkVisibility` is named `networking.knative.dev/visibility` . We could have named it more kserve specific as knative is not even in picture if RawDeployment mode is used. This key's name can be confusing.
Some other options would be:
`networking.kserve.dev/visibility`